be/src/exprs/vectorized/arithmetic_expr.cpp

```diff
@@ -18,11 +21,68 @@ namespace starrocks::vectorized {
                                                       \
     virtual Expr* clone(ObjectPool* pool) const override { return pool->add(new CLASS_NAME(*this)); }
 
+static std::optional<PrimitiveType> eliminate_trivial_cast_for_decimal_mul(const Expr* e) {
```

Comment:
may be we could define a function in FE ? Mul(decimal32, decimal128) ?

Reply:
It is not required, we can consider its an abstract leak to optimize general-purpose multiplication.